### DISCUSSION TITLE
How can I prevent useMutation from re-rendering my component ?

### DISCUSSION THREAD

**Discussion by gkpo** [urql-graphql/urql#2912] [Q&A] [answered]
I have a component with a mutation for which I don't care about the results. So there is no need to re-render the component after the mutation is executed.

Example:

```
const UpdateDateAppLastInteracted = `
  mutation UpdateDateAppLastInteracted($currentDateTime: timestamptz) {
    ...
  }
`;

const App = ({ id, title }) => {
  const [, doUpdateAppLastInteracted] = useMutation(UpdateDateAppLastInteracted);

  console.log("App rendered");
  
  const onRouteChange = () => {
    doUpdateAppLastInteracted({currentDateTime: new Date()});
  }

  // Call onRouteChange whenever the route changes
};
```

My problem here is that calling the mutation results in the component being re-rendered. How can I tell urql that I don't care about the results and I don't want to re-render?

Thanks in advance

**Comment by kitten** [comment, marked as answer]
First of all, I wouldn't want to worry about re-renders in my opinion. Re-renders are a frequent and "natural" occurrence in React, and by design I wouldn't want to prevent all of them, and instead, let React bail out as it sees fit, or prevent them selectively (`React.memo` is an easy barrier to add for expensive components for instance.)

That said, if you _really_ don't care about the results, I would get the client manually and then fire off the mutation programmatically: https://formidable.com/open-source/urql/docs/basics/core/#one-off-queries-and-mutations
We call this "one-off mutations" and it can be triggered using `client.mutation(...).toPromise()`. This will send off the mutation and give you a promise result as soon as you call `toPromise` on the source.

```js
import { useClient } from 'urql';

const Component = () => {
  const client = useClient();

  const onRouteChange = () => {
    client.mutation(UpdateDateAppLastInteracted, { currentDateTime: new Date() }).toPromise();
  };

  // ...
};
```

**Reply by gkpo** [reply]
Thanks a lot!
